warehouse/query-core/src/main/java/datawave/query/jexl/DatawaveArithmetic.java

```diff
@@ -231,6 +236,91 @@ protected Long subtract(Collection<?> left, Collection<?> right) {
         return null;
     }
     
+    protected void addAll(Collection<Object> set, Object o) {
+        if (o instanceof Collection) {
+            set.addAll((Collection<?>) o);
+        } else {
+            set.add(o);
+        }
+    }
+    
+    /**
+     * This method deals with the ValueTuple objects and turns them into the normalized value parts
+     *
+     * @param o
+     * @return
+     */
+    protected Object normalizedValues(Object o) {
+        if (o instanceof Set) {
+            Set<Object> normalizedValues = new HashSet<>();
+            for (Object value : ((Set<?>) o)) {
+                addAll(normalizedValues, normalizedValues(value));
+            }
+            if (normalizedValues.size() == 1) {
+                return normalizedValues.iterator().next();
+            } else {
+                return normalizedValues;
+            }
+        } else {
+            return ValueTuple.getNormalizedValue(o);
+        }
+    }
+    
+    /**
+     * Tests whether the right value matches our FST.
+     *
+     * @param fst
+     *            first value
+     * @param right
+     *            second value
+     * @return test result.
+     */
+    public boolean fstMatch(FST fst, Object right) {
+        right = normalizedValues(right);
+        
+        if (right instanceof Set) {
+            Set<Object> set = (Set<Object>) right;
+            
+            for (Object o : set) {
+                if (o != null) {
+                    final IntsRefBuilder irBuilder = new IntsRefBuilder();
+                    Util.toUTF16(o.toString(), irBuilder);
+                    final IntsRef ints = irBuilder.get();
+                    synchronized (fst) {
+                        try {
+                            if (Util.get(fst, ints) != null) {
+                                return true;
+                            }
+                        } catch (IOException e) {
+                            if (log.isTraceEnabled()) {
+                                log.trace("Failed to evaluate " + o.toString() + " against the FST.");
+                            }
+                        }
+                    }
+                }
+            }
+        } else {
+            if (right != null) {
+                final IntsRefBuilder irBuilder = new IntsRefBuilder();
```

Comment:
this could be a function to avoid the code copy/paste

Reply:
I agree.  Updated.